python/python/lance/torch/data.py

```diff
@@ -150,6 +150,9 @@ def __init__(
         rank: Optional[int] = None,
         world_size: Optional[int] = None,
         shard_granularity: Optional[Literal["fragment", "batch"]] = "fragment",
+        to_tensor_fn: callable[
+            [pa.RecordBatch], Union[dict[str, torch.Tensor], torch.Tensor]
+        ] = _to_tensor,
```

Comment:
It looks like this can be `Union[callable[...], None]`?

Reply:
Sure i can make it an `Option[callable]`,